contracts/RandomNumberConsumer.sol

```diff
@@ -17,22 +18,26 @@ contract RandomNumberConsumer is VRFConsumerBase {
      * LINK token address:                0xa36085F69e2889c224210F603D836748e7dC0088
      * Key Hash: 0x6c3699283bda56ad74f6b855546325b68d482e983852a7a82979cc4807b641f4
      */
-    constructor(address _linkTokenAddress)
+    constructor(address _linkTokenAddress, bytes32 _keyHash, 
+    address _vrfCoordinatorAddress, uint256 _fee)
         public
         VRFConsumerBase(
-            0xdD3782915140c8f3b190B5D67eAc6dc5760C46E9, // VRF Coordinator
+            _vrfCoordinatorAddress, // VRF Coordinator
             _linkTokenAddress  // LINK Token
         )
     {
-        keyHash = 0x6c3699283bda56ad74f6b855546325b68d482e983852a7a82979cc4807b641f4;
-        fee = 0.1 * 10 ** 18; // 0.1 LINK
+        keyHash = _keyHash;
+        fee = _fee;
+        // fee = 0.1 * 10 ** 18; // 0.1 LINK
     }
     
     /** 
      * Requests randomness from a user-provided seed
      */
     function getRandomNumber(uint256 userProvidedSeed) public returns (bytes32 requestId) {
-        return requestRandomness(keyHash, fee, userProvidedSeed);
+        requestId = requestRandomness(keyHash, fee, userProvidedSeed);
+        emit RequestedRandomness(requestId);
+        return requestId;
```

Comment:
Cool thing about Solidity is since you're defining the return variable above (`bytes32 requestId`) this return line is redundant and you can remove it.

Reply:
Oh sweet. But isn't that less readable?